modules/iam-compartment/variables.tf

```diff
@@ -8,6 +8,11 @@ variable "compartment_name" {
   description = "The name you assign to the compartment during creation. The name must be unique across all compartments in the tenancy. "
 }
 
+variable "compartment_id" {
+  description = "The OCID of the parent compartment containing the compartment. Allow for sub-compartments creation"
+  default     = ""
```

Comment:
Done.

- Module iam-compartment variable is initialized to null,
- Module iam-compartment resource's argument "compartment_id" is made conditional.

If compartment_id is not explicitly provided, it will default to tenancy_id value (hence, create the compartment at root level).